Docs 7161 edit (#602)
mulesoft/docs-connectors: modules/ROOT/pages/email/email-list.adoc

include::_attributes.adoc[]
endif::[]

> Email Connector can list emails from any part of the flow. This capability is a change from the Mule 3 transport, which could only retrieve emails as a result of inbound endpoint polling.

Reviewer: ditto re. emails; if you change, change GLOBALLY

> Both the POP3 and IMAP configurations contain a `list` operation. This section covers both and explains their differences. Though the following examples might use `list-pop3` or `list-imap`, all of them apply to both configurations unless stated otherwise.

Reviewer: So...we want to create reusability by not referring to "this section" or "the following" when the following is under a separate title.
There are many ways to work around this, but here is the first that came to my mind. You will likely think of something much better!
```suggestion
Both the POP3 and IMAP configurations contain a `list` operation. Although there are differences between the configurations, the examples using either `list-pop3` or `list-imap` apply to both configurations unless stated otherwise.
```
However you change it, be sure to replace "though" with "although."

=== Secured Connections

> The IMAP and POP3 configurations provide a secured connection type to work over the secured versions of the protocols (POP3S and IMAPS). These connection types allow you to configure a TLS context that enables SSL/TLS encryption.

Reviewer: The first sentence sounds redundant to me, but I am not quite sure I have an alternative that would have the exact same meaning. How's this?
```suggestion
The IMAP and POP3 configurations enable connections that are secured by the secured versions of the POP3S and IMAPS protocols. These connections enable you to configure a TLS context that enables both SSL and TLS encryption.
```

<email:list-imap config-ref="imap-config" mailboxFolder="INBOX"/>
----

> The previous examples return all the emails in the configured mailbox servers that reside in the INBOX folder.

Reviewer: Does this say that the email messages are in the servers, and the servers are in the INBOX folder?
Is it supposed to say that the email messages are in the INBOX folders of the servers?
I'm just a little confused about what is in what else: mail > folder > server makes the most logical sense to me, but that doesn't mean it's correct in this context.

When listing emails, you might be interested not only in the email's content but also in its metadata (for example, the subject, from addresses, to addresses, received date, and so on). These attributes vary depending on the type of configuration that you are using. For example, the IMAP protocol provides more metadata about the retrieved email like the `recent`, `seen`, `deleted`, and `answered` flags.

Email Connector uses Mule message attributes to access this information, while the message payload always contains the email's content.

For details on the structure of email attributes, see  xref:email/email-documentation.adoc[Email Module Documentation Reference].

== Matching

> When listing emails, you can filter emails that match certain criteria using the `<email:pop3-matcher>` and `<email:imap-matcher>` elements. These elements define the criteria that can be used to process an email.

Reviewer: ```suggestion
When listing emails, you can filter them using the `<email:pop3-matcher>` and `<email:imap-matcher>` elements. These elements define the criteria that can be used to process an email.
```
"that can be used" or "that is used"? "that you can set to"

All matcher attributes are optional and are ignored if they are not provided. They are also related to each other under an AND operator, meaning that all the criteria must be true.

> The previous example declares an inner, non-reusable matcher that is proprietary to a particular component. However, matchers can be declared as a reusable element (as a named top-level element) like the following `pop3-matcher`:

Reviewer: ```suggestion
The previous example declares an inner, non-reusable matcher that is proprietary to a particular component. However, matchers can be declared as reusable elements (as named top-level elements), such as the following `pop3-matcher`:
```

=== IMAP Matcher versus POP3 Matcher

> The IMAP protocol provides metadata about the email that allows for more precise filters than POP3.

Reviewer: ```suggestion
The IMAP protocol provides metadata about the email that allows more precise filtering than POP3.
```

  answered="EXCLUDE|INCLUDE|REQUIRE"/>
----

Notice that the IMAP matcher includes the `recent`, `seen`, `deleted`, and `answered` parameters.

> == Naming Strategy For Attachments

Reviewer: ```suggestion
== Naming Strategy for Attachments
```

> Email Connector provides a parameter to specify a strategy for how to name attachments for the List and On New Email operations. This parameter is set at the configuration level, but can be overridden at the operation

Reviewer: "provides a parameter to specify a strategy for how to name attachments" makes my head hurt, although it's not wrong. Oh. This is the thing from your Slack message, right? What I said there.

level. 

See xref:email/email-trigger.adoc[Naming Strategy For Attachments] for more information and examples.

== See Also

* xref:email/email-send.adoc[Sending Emails with the Email Connector]

> * xref:email/email-trigger.adoc[Triggering a Flow When a New Email is Received]

Reviewer: ```suggestion
* xref:email/email-trigger.adoc[Triggering a Flow When a New Email Is Received]
```
